lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -136,10 +145,18 @@ def script
           # Strip invalid architectures
           strip_invalid_archs() {
             binary="$1"
-            # Get architectures for current file
-            archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | rev)"
+            # Get architectures for current target binary
+            binary_archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | awk '{$1=$1;print}' | rev)"
+            # Intersect them with the architectures we are building for
+            intersected_archs="$(echo ${ARCHS[@]} ${binary_archs[@]} | tr ' ' '\\n' | sort | uniq -d)"
+            # If there are no archs supported by this binary then warn the user
+            if [[ -z "$intersected_archs" ]]; then
+              echo "warning: [CP] Vendored binary '$binary' contains architectures ($binary_archs) none of which match the current build architectures ($ARCHS)."
```

Comment:
using `warning:` will actually show a warning in Xcode.

Reply:
so in order to avoid the problem i should upgrade cocoapods ?